### PR TITLE
Sequence player

### PR DESCRIPTION
**Pull request by marcharper**

### CODE REVIEW

File: axelrod/strategies/sequence_player.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    @init_args
+    def __init__(self, generator_function, generator_args=None):
+        Player.__init__(self)
+        # Initialize the sequence generator
+        self.generator_function = generator_function
+        self.generator_args = generator_args
+        self.sequence_generator = self.generator_function(*self.generator_args)
+
+    def meta_strategy(self, value):
```
Comment (drvinceknight) [inline axelrod/strategies/sequence_player.py]: Nice. I was wondering what would be the best way to do this.

File: axelrod/strategies/sequence_player.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        for s in self.sequence_generator:
+            return self.meta_strategy(s)
+
+    def reset(self):
+        # Be sure to reset the sequence generator
+        Player.reset(self)
+        self.sequence_generator = self.generator_function(*self.generator_args)
+
+
+class Memoized(object):
```
Comment (drvinceknight) [inline axelrod/strategies/sequence_player.py]: Any reason not to use this: https://docs.python.org/3/library/functools.html#functools.lru_cache

(Saw that in a talk the other day, I have no experience with either.)

Comment (marcharper) [inline axelrod/strategies/sequence_player.py]: It's python 3 only :/

Comment (drvinceknight) [inline axelrod/strategies/sequence_player.py]: That'll be why then :) :+1:

**Comment by drvinceknight** [comment]
@geraintpalmer I suggest you merge this in and pass it on via your original PR.

Could be great to include tests for the parent sequence player (it's very generic so perhaps not everything is testable).

**Comment by marcharper** [comment]
It should update the original PR once merged.

File: axelrod/strategies/sequence_player.py (hunk 2 above)
Comment (marcharper) [inline axelrod/strategies/sequence_player.py]: More reasons every day to leave Python 2 behind!

Comment (drvinceknight) [inline axelrod/strategies/sequence_player.py]: This is certainly one of the best ones I've seen.

**Pull request merged**